actionview/lib/action_view/helpers/url_helper.rb

```diff
@@ -172,6 +172,25 @@ def _filtered_referrer # :nodoc:
       #   link_to "External link", "http://www.rubyonrails.org/", target: "_blank", rel: "nofollow"
       #   # => <a href="http://www.rubyonrails.org/" target="_blank" rel="nofollow">External link</a>
       #
+      # ==== Turbo
+      #
+      # Rails 7 ships with Turbo enabled by default. The <tt>data-turbo-method</tt> HTML attribute can be used
+      # to perform a Turbo link visit with a different HTTP method.
+      #
+      # A common use case is to insert links to perform update or delete operations (i.e. non-<tt>GET</tt> requests) in a form.
+      # For example, you might use +link_to+ in an edit form to delete a record:
+      #
+      #   link_to "Delete profile", @profile, data: { turbo_method: :delete }
+      #   # => <a href="/profiles/1" data-turbo-method="delete">Delete profile</a>
+      #
+      # Since forms cannot appear inside another form, +button_to+ <i>cannot</i> be used here as it returns a form.
```

Comment:
Hmm. Doesn't `data-turbo-method` also generate a form at runtime?

Reply:
Great question, @ghiculescu ! 

`data-turbo-method` _does_ generate a form at runtime. But, [it appends the generated form as a child of `document.body`](https://github.com/hotwired/turbo/blob/f8b0e17a0c605080de0097112a5cf408abef9bb1/src/observers/form_link_click_observer.ts#L63) instead of it returning it directly like `button_to`, which is a neat shortcut to make non-`GET` requests work in forms.

[The Turbo handbook does mention this](https://turbo.hotwired.dev/handbook/drive#performing-visits-with-a-different-method):

> The link will get converted into a hidden form next to the a element in the DOM. This means that the link can’t appear inside another form, as you can’t have nested forms.

I'm not entirely sure if that's accurate since this type of link works in my testing. The source code also suggests that the form is appended to `document.body` rather than the link where `data-turbo-method` is called.

Reply:
> I'm not entirely sure if that's accurate since this type of link works in my testing. The source code also suggests that the form is appended to document.body rather than the link where data-turbo-method is called.

Should probalby update the Turbo handbook.

Reply:
Good idea, @ghiculescu ! I will create a PR shortly for updating the Turbo handbook regarding `data-turbo-method`.

Reply:
Opened in: https://github.com/hotwired/turbo-site/pull/142